internal/server/statvar/statvar_search.go

```diff
@@ -187,14 +175,27 @@ func groupStatVars(
 	for _, svg := range svgMap {
 		resultSvg = append(resultSvg, svg)
 	}
+	// Sort the stat var group results
+	sort.SliceStable(resultSvg, func(i, j int) bool {
+		svgi := resultSvg[i]
+		svgj := resultSvg[j]
+		if len(svgi.StatVars) != len(svgj.StatVars) {
+			return len(svgi.StatVars) > len(svgj.StatVars)
+		}
+		ri := ranking[resultSvg[i].Dcid]
+		rj := ranking[resultSvg[j].Dcid]
+		return compareRankingInfo(ri, resultSvg[i].Dcid, rj, resultSvg[j].Dcid)
+	})
 	return resultSv, resultSvg
 }
 
 func searchTokens(
 	tokens []string, index *resource.SearchIndex,
 ) ([]*pb.EntityInfo, []*pb.EntityInfo, []string) {
-	svMatches := map[string][]string{}
-	svgMatches := map[string][]string{}
+	// svMatches and svgMatches are maps of sv/svg id to a map of tokens that
+	// match each sv/svg to a set of strings that match the token
+	svMatches := map[string]map[string]map[string]struct{}{}
```

Comment:
maybe have a new type def for at lease 1/2 of the map here, so it is easy to see what's the map is. That also simplify the init statements below.

Reply:
+1, it's also worth to do a typedef for empty = struct{}{} for readability

Reply:
added new type def for part of the map.

but not quite understanding where the empty = struct{}{} will be used? Because I'm not setting anything to struct{}{}

Reply:
ah ok, i was misreading that statement. i thought it was an assignment